src/commands.rs

```diff
@@ -89,6 +89,13 @@ fn run_server_process(startup_timeout: Option<Duration>) -> Result<ServerStartup
     let runtime = Runtime::new()?;
     let exe_path = env::current_exe()?;
     let workdir = exe_path.parent().expect("executable path has no parent?!");
+
+    // Spawn a blocking task to bind the Unix socket. Note that the socket
+    // must be bound before spawning `_child` below to avoid a race between
+    // the parent binding the socket and the child connecting to it.
+    let sp = socket_path.clone();
+    let listener = runtime.block_on(async move { tokio::net::UnixListener::bind(sp) })?;
```

Comment:
This can be written `let listener = tokio::net::UnixListener::bind(&socket_path.clone())?;`

Reply:
It compiles, but when I run it I get:
```
$ ./target/debug/sccache --start-server
sccache: Starting the server...
thread 'main' panicked at 'there is no reactor running, must be called from the context of a Tokio 1.x runtime', src/commands.rs:98:20
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The way I understand it: the function `run_server_process` is not async. Therefore, you cannot call async functions in it without starting a Tokio runtime(?). `tokio::net::UnixListener::bind` is an async function. Therefore, you must call it inside a `runtime.block_on` call so that it runs in a runtime(?).

Reply:
The function is not, actually, async. https://docs.rs/tokio/latest/tokio/net/struct.UnixListener.html#method.bind But it does not want to run without a runtime.
So the following would work too, without an async block:
```
let listener = {
    let _guard = runtime.enter();
    tokio::net::UnixListener::bind(&socket_path.clone())?
};
```

Reply:
Indeed. I've pushed a fixup with your latest suggestion.